.github/workflows/bazel-lockfile.yml

```diff
@@ -0,0 +1,222 @@
+# Workflow to automatically update the bazel lockfile on pull requests and dev
+# branches. In general, fails if lockfile is out of date.
+# For each platform we support, we run a dry (--nobuild) Bazel build, and then
+# upload the resulting lockfile as an artifact. The next platform downloads the
+# lockfile artifact, and then also runs a dry build, rinse, repeat.
+name: Update MODULE.bazel.lock
+on:
+  # Merge groups should be checked for up-to-date lockfile
+  merge_group:
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+  pull_request:
+  push:
+    branches:
+      # master is checked for up-to-date lockfile
+      - master
+      # dev branches get a new commit with updated lockfile (when necessary)
+      - 'dev-gh-*'
+
+# runs for the same workflow are cancelled on PRs but not on master
+# (logic copied from main workflow)
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  bazel-bzlmod-lockfile-apple-silicon:
+    name: Apple Silicon
+    runs-on: namespace-profile-darwin # profile created in namespace console
+    steps:
+      - uses: actions/checkout@v4
+      - name: Bazel dry run
+        run: |
+          bazel \
+            --noworkspace_rc \
+            --bazelrc=./bazel/conf/.bazelrc.build \
+            build \
+            --config=ci --config=macos_ci \
+            --test_tag_filters="test_macos,test_macos_slow,-upload" \
+            //rs/... //publish/binaries/... \
+            --lockfile_mode=update \
+            --nobuild
+      - uses: actions/upload-artifact@v4
+        with:
+          name: bazel-module-lock-apple-silicon
+          path: ./MODULE.bazel.lock
+
+  bazel-bzlmod-lockfile-linux:
+    name: Linux
+    container:
+      image: ghcr.io/dfinity/ic-build@sha256:115daa5ad5149182bb0416cbe5730f305be3bb2f48df576bc2c23067eefce84b
+      options: >-
+        -e NODE_NAME
```

Comment:
I see it should be `-e NODE_NAME -v /cache:/cache` since `--repository_cache=/cache/bazel` is used.